### ISSUE TITLE
NullPointerException

### ISSUE THREAD

**Issue by Rakambda** [om3g4zell/CityBuilderJSFML#6]
Sometimes when exiting the drawing mode, the game crashes.

``` java
Exception in thread "main" java.lang.NullPointerException
    at sim.Sim.getBuildingsOfType(Sim.java:577)
    at sim.Sim.spawnBuildings(Sim.java:664)
    at sim.Sim.update(Sim.java:1024)
    at sim.Main.main(Main.java:72)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke(Method.java:498)
    at com.intellij.rt.execution.application.AppMain.main(AppMain.java:144)
```

**Comment by Qu3tzal** [comment]
I can't reproduce the bug and the lines indicated in your stack dump don't match the methods.
Are you sure you have the last version ?

**Comment by Rakambda** [comment]
It's very random, but happened already several times.

For the lines, it must be because I changed lines to be able to make the res folder work with a thing other than Eclipse: [see here](https://github.com/MrCraftCod/CityBuilderJSFML/blob/master/CityBuilderJSFML/src/sim/Sim.java#L577)

**Comment by Qu3tzal** [comment]
Alright, should be corrected with commit f620d1d4e5cbed473ca48796343d7d9eb39ffd89.
I can't figure out why `b` would sometimes be `null`. It seems like a ghost building happens to require the creation of a building.
Anyway, thank you !
